client/state/automated-transfer/reducer.js

```diff
@@ -35,6 +35,10 @@ export const status = withPersistence( ( state = null, action ) => {
 			return action.status;
 		case TRANSFER_UPDATE:
 			return 'complete' === action.status ? transferStates.COMPLETE : state;
+		case REQUEST_STATUS_FAILURE:
+			return action.error === 'An invalid transfer ID was passed.'
+				? transferStates.NONE
+				: transferStates.REQUEST_FAILURE;
```

Comment:
It's not clear to me why we would want to distinguish these two cases. Could you help me understand more about this?

It stands out to me because relying on a human-readable error message string like this is often fragile. However, I also don't see anything like a better error code from the endpoint response. Thus, if we can simplify this case as just a REQUEST_FAILURE or the other way around, then we could save some potential maintenance headache.

Reply:
Actually there is no way for us to figure out if a transfer exists or not. The only way to distinguish between a failed request and a non existent transfer is this error message. This is essential since when we fetch for the transfer details initially if there is no transfer there is no meaningful message returned. A better way is to possibly maintain a proper DOES_NOT_EXIST transfer status in the backend as well. 

Can you think of a better approach for this? Addition to the above I feel that tagging a non existent transfer as a request failure is extremely confusing. But do agree that its dangerous to rely on an error message like this. Maybe a backend change is in order :)

Reply:
I agree that relying on the error message here can be fragile. Especially in multilingual environments - what will happen if the message is localized and displayed in another language? 

I agree that this looks like a signal for a backend change to add better support for differentiating between the different failure types.

Reply:
Will be working on this in a subsequent PR : tr-c/KfPtazSp